ci/run_docker.sh

```diff
@@ -2,13 +2,13 @@
 
 set -e
 
-BUILD_IMAGE=opentelemetry-cpp-build
+BUILD_IMAGE=opentelemetry-cpp-build-test
+
 docker image inspect "$BUILD_IMAGE" &> /dev/null || {
   docker build -t "$BUILD_IMAGE" ci
```

Comment:
It's probably adding considerable amount of CI time to build an image. Have you considered running the ci scripts without a docker, to deploy the tools in one of the readily available GitHub Action images?
https://raw.githubusercontent.com/maxgolov/opentelemetry-cpp/master/.github/workflows/build-ubuntu.yml
Main idea is that total compute and run time for each PR is going to be less, since you won't need to build an image.